client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -413,13 +392,25 @@ public void request(long l) {
                 throw new IllegalStateException("request() must be called from the event loop managing the channel.");
             }
 
-            chunksRequested += l;
-            while (!queuedContent.isEmpty() && chunksRequested > 0 && !isCanceled) {
-                emitContent(queuedContent.remove());
-            }
+            if (l <= 0) {
+                subscriber.onError(new IllegalArgumentException("Request amount must be positive. Received amount: " + l));
+            } else {
+                chunksRequested = BackpressureHelper.addCap(chunksRequested, l);
+
+                if (!draining) {
```

Comment:
This drain section should be in its own method and called by `request` and also called by `onReceivedContent` otherwise the stream can stall.